### PR TITLE
Augmented assignments to WCR

### PR DESCRIPTION
**Pull request by tbennun**

### CODE REVIEW

**Review by definelicht — APPROVED** [review]

**Comment by definelicht** [comment]
@tbennun Can you review? I cannot request you, because it's your own PR :-)

File: dace/sdfg/validation.py [hunk 1]
```diff
@@ -481,14 +481,15 @@ def validate_state(state: 'dace.sdfg.SDFGState',
...
-                )
+                pass
+                # raise InvalidSDFGEdgeError(
+                #     "Memlet creates an "
+                #     "invalid path (sink node %s should "
+                #     "be a data node)" % str(dst_node),
+                #     sdfg,
+                #     state_id,
+                #     eid,
+                # )
```
Comment (tbennun) [inline dace/sdfg/validation.py:492]: well, now you're allowing a bad pattern to validate:
```
<tasklet>
    |
    v
\mapexit/
    |
    v
<tasklet>
```

Comment (tbennun) [reply dace/sdfg/validation.py:492]: The difference here should be that we can end in an `nd.ExitNode` if and only if the `edge.data.is_empty()`

File: dace/transformation/dataflow/wcr_conversion.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+from dace import registry, nodes, dtypes
+from dace.transformation import transformation
+from dace.sdfg import utils as sdutil
+from dace import SDFG, SDFGState
+
+
+@registry.autoregister_params(singlestate=True)
+class AugAssignToWCR(transformation.Transformation):
+    """
+    Converts a augmented assignment ("a += b", "a = a + b") into a tasklet
```
Comment (tbennun) [inline dace/transformation/dataflow/wcr_conversion.py:13]:
```suggestion
    Converts an augmented assignment ("a += b", "a = a + b") into a tasklet
```

**Review by tbennun — COMMENTED** [review]
@definelicht validation needs to change a bit, otherwise some minor fixes

File: dace/sdfg/validation.py (hunk 1 above)
Comment (definelicht) [reply dace/sdfg/validation.py:492]: Makes sense! tbh I just didn't know what the right condition was, so I put this in the PR to trigger your input :kappa:

**Comment by definelicht** [comment]
@tbennun Requesting review again, please pay special attention to the changes to `read_and_write_sets` in both `sdfg.py` and `state.py`

**Review by definelicht — APPROVED** [review]

**Pull request merged**